File: python_tests/utils/config_file.py
import inspect
import yaml
import os
from typing import Any, Dict


def read_config() -> Dict[str, Any]:
    caller_frame = inspect.stack()[1]
    file_path = os.path.abspath(caller_frame.filename)
    current_directory = os.path.dirname(file_path)
    config_file = os.path.join(current_directory, "config.yaml")

    with open(config_file, "r") as file:
        return yaml.safe_load(file)


def write_config(data: Dict[str, Any]) -> None:
    caller_frame = inspect.stack()[1]
    file_path = os.path.abspath(caller_frame.filename)
    current_directory = os.path.dirname(file_path)
    config_file = os.path.join(current_directory, "config.yaml")
    with open(config_file, "w+") as file:
        yaml.safe_dump(data, file)

File: go_tests/utils/config_file.go
package utils

import (
	"fmt"
	"os"
	"path/filepath"
	"runtime"

	"gopkg.in/yaml.v3"
)

func ReadConfig() (data map[string]any, err error) {
	_, callerFile, _, ok := runtime.Caller(1)
	if !ok {
		return nil, fmt.Errorf("could not get caller information")
	}

	currentDirectory := filepath.Dir(callerFile)
	parentDirectory, err := filepath.Abs(filepath.Join(currentDirectory, ".."))
	if err != nil {
		return nil, fmt.Errorf("could not get parent directory: %w", err)
	}

	configFile := filepath.Join(parentDirectory, "config.yaml")

	fileContent, err := os.ReadFile(configFile)
	if err != nil {
		return nil, err
	}

	err = yaml.Unmarshal(fileContent, &data)
	if err != nil {
		return nil, err
	}

	return data, nil
}

func WriteConfig(data map[string]any) (err error) {
	_, callerFile, _, ok := runtime.Caller(1)
	if !ok {
		return fmt.Errorf("could not get caller information")
	}

	currentDirectory := filepath.Dir(callerFile)
	configFile := filepath.Join(currentDirectory, "config.yaml")

	fileContent, err := yaml.Marshal(data)
	if err != nil {
		return err
	}

	err = os.WriteFile(configFile, fileContent, 0644)
	if err != nil {
		return err
	}

	return nil
}
